pkg/platform/api/api.go

```diff
@@ -0,0 +1,201 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/bottlerocket-os/bottlerocket-update-operator/pkg/logging"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+)
+
+const bottlerocketAPISock = "/run/api.sock"
+const minimumRequiredOSVer = "0.3.4"
+
+type updateState = string
+
+const (
+	Idle      updateState = "Idle"
+	Available updateState = "Available"
+	Staged    updateState = "Staged"
+	Ready     updateState = "Ready"
+)
+
+type UpdateImage struct {
+	Arch    string `json:"arch"`
+	Version string `json:"version"`
+	Variant string `json:"variant"`
+}
+
+func (ui *UpdateImage) Identifier() interface{} {
+	return ui.Version
+}
+
+type StagedImage struct {
+	Image      UpdateImage `json:"image"`
+	NextToBoot bool        `json:"next_to_boot"`
+}
+
+type updateCommand = string
+
+const (
+	refresh    updateCommand = "refresh"
+	prepare    updateCommand = "prepare"
+	activate   updateCommand = "activate"
+	deactivate updateCommand = "deactivate"
+)
+
+type commandStatus = string
+
+const (
+	Success commandStatus = "Success"
+	Failed  commandStatus = "Failed"
+	Unknown commandStatus = "Unknown"
+)
+
+type CommandResult struct {
+	CmdType    updateCommand `json:"cmd_type"`
+	CmdStatus  commandStatus `json:"cmd_status"`
+	Timestamp  string        `json:"timestamp"`
+	ExitStatus *int32        `json:"exit_status"`
+	Stderr     *string       `json:"stderr"`
+}
+
+type UpdateStatus struct {
+	UpdateState       updateState    `json:"update_state"`
+	AvailableUpdates  []string       `json:"available_updates"`
+	ChosenUpdate      *UpdateImage   `json:"chosen_update"`
+	ActivePartition   *StagedImage   `json:"active_partition"`
+	StagingPartition  *StagedImage   `json:"staging_partition"`
+	MostRecentCommand *CommandResult `json:"most_recent_command"`
+}
+
+type APIClient struct {
+	log logging.Logger
+	httpClient *http.Client
+}
+
+func NewAPIClient() *APIClient {
+	return &APIClient {log: logging.New("api"), httpClient: &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
+				dialer := net.Dialer{}
+				return dialer.DialContext(ctx, "unix", bottlerocketAPISock)
+			},
+		},
+		// Set a 10 second timeout for all requests
+		Timeout: 10 * time.Second,
+	},
+	}
+}
+
+// makeAPIRequest makes an http request to the Bottlerocket API and returns the response
+// We only retry the request if the response was a 423 Locked (indicates that the update API is not ready)
+func(c *APIClient) makeAPIRequest(method string, path string) (*http.Response, error) {
+	var response *http.Response
+	var attempts = 0
```

Comment:
```suggestion
	attempts := 0
```

Idiomatically, use `var` when you specifically need its separation of declaration and initialization.  If you're initializing on the same line (and not overriding the type), generally `:=` is preferred.